src/test/scala/org/constellation/domain/transaction/TransactionChainServiceTest.scala

```diff
@@ -37,17 +39,145 @@ class TransactionChainServiceTest
       last.unsafeRunSync shouldBe LastTransactionRef("", 0L)
     }
 
-    "should return last transaction ref if address is known" in {}
+    "should return last transaction ref if address is known" in {
+      val address = "DAG123"
+      val lastTx = LastTransactionRef("tx123", 1)
+      service.lastTransactionRef.set(Map(address -> lastTx)).unsafeRunSync
+
+      val last = service.getLastTransactionRef(address).unsafeRunSync
+      last shouldEqual lastTx
```

Comment:
if you're writing such async tests, IMO it's more readable to chain it:
```
val init = service.lastTxRef.set(..)
val getLast = service.getLastTxRef...()

(init >> getLast).unsafeRunSync shouldBe lastTx
```

Reply:
Alright! I will try that!